src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborWriterTests.Integer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    public partial class CborWriterTests
+    {
+        // Data points taken from https://tools.ietf.org/html/rfc7049#appendix-A
+
+        [Theory]
+        [InlineData(0, "00")]
+        [InlineData(1, "01")]
+        [InlineData(10, "0a")]
+        [InlineData(23, "17")]
```

Comment:
If I recall correctly, this can also be encoded as `1817`, `190017` , `1a00000017` and `1b0000000000000017`.  We should add tests that show that we can read those, or refuse to read those, whichever is appropriate.

Reply:
It depends on whether we choose to enforce canonical cbor, c.f. https://tools.ietf.org/html/rfc7049#section-3.9

Reply:
It feels like both this behaviour and utf8 error tolerance could be exposed as configuration. But it really depends on requirements, effort, and whether we think this is essential for a binary format.

Reply:
Supporting the reader enforcing canonical mode is on the list of feature requirements (toward the end), and presumably someone will want a lax mode.

But as a binary format reader, particularly one used as part of making trust decisions, there can't be any surprises.  If something is possible, we want a test for it.  Whether to codify we don't support it, or we do, there needs to be a test for it.